src/main/java/org/junit/experimental/theories/internal/Assignments.java

```diff
@@ -85,14 +85,16 @@ public Assignments assignNext(PotentialAssignment source) {
 
     private List<PotentialAssignment> generateAssignmentsFromTypeAlone(ParameterSignature unassigned) {
         Class<?> paramType = unassigned.getType();
-        
-        if (paramType.isEnum()) {
-            return new EnumSupplier(paramType).getValueSources(unassigned);  
-        } else if (paramType.equals(Boolean.class) || paramType.equals(boolean.class)) {
-            return new BooleanSupplier().getValueSources(unassigned);
-        } else {
-            return emptyList();
+
+        FromDataPoints fromDataPoints = unassigned.getAnnotation(FromDataPoints.class);
+        if (fromDataPoints == null) {
+            if (paramType.isEnum()) {
+                return new EnumSupplier(paramType).getValueSources(unassigned);
+            } else if (paramType.equals(Boolean.class) || paramType.equals(boolean.class)) {
+                return new BooleanSupplier().getValueSources(unassigned);
+            }
         }
+        return emptyList();
```

Comment:
For symmetry it would be nice to keep the `else`

```suggestion
        } else {
            return emptyList();
        }
```